### ISSUE TITLE
[VMware_guest] Error with OS_customization when not trying to.

### ISSUE THREAD

**Issue by BarresB** [ansible/ansible#41736]
##### SUMMARY
When executing the playbook I get an error that I can't customise the OS. That's is correct, but looking at the playbook i'm not trying to. 

I wouldn't mind so much, but the machine isn't running after the playbook. I can work around that by an extra task that only powers on the machine. But I don't want that, plus the message that it has failed is unwanted of course.

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
vmware_guest

##### ANSIBLE VERSION
```
ansible 2.5.4
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/ansible/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Feb 20 2018, 09:19:12) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
```

##### OS / ENVIRONMENT
Ansible running on RHEL7.5
trying to configure a vmware template of Pulse PCS 8.3R4
vCenter 6.7.0
ESXi host 6.0

##### STEPS TO REPRODUCE

```yaml
- hosts: 192.168.23.40
  gather_facts: no
  tasks:
    - name: Create a PulseConnectSecureVM from the template
      vmware_guest:
       hostname: 192.168.23.40
       username: administrator@vsphere.local
       password: password
       datacenter: Datacenter
       validate_certs: no
       folder: /bla/bla/
       name: PulseConnectSecure
       state: poweredon
       template: Pulse Connect Secure
       disk:
       - size_gb: 40
         datastore: datastore1
       networks:
       #internal
       - name: VM Network
       #management
       - name: VM Network 2
       #external
       - name: VM Network
      delegate_to: localhost
      register: deploy
```

##### EXPECTED RESULTS
Machine deployed and running.

##### ACTUAL RESULTS
Machine is deployed, but not running. Manually starting the machine seems to work fine and I can't discover any problems.

```
2018-06-20 14:54:00,313 p=23299 u=ansible |  ansible-playbook 2.5.4
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/ansible/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.5 (default, Feb 20 2018, 09:19:12) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
2018-06-20 14:54:00,313 p=23299 u=ansible |  Using /etc/ansible/ansible.cfg as config file
2018-06-20 14:54:00,313 p=23299 u=ansible |  setting up inventory plugins
2018-06-20 14:54:00,324 p=23299 u=ansible |  Parsed /home/ansible/inventory.ini inventory source with ini plugin
2018-06-20 14:54:00,400 p=23299 u=ansible |  Loading callback plugin default of type stdout, v2.0 from /usr/lib/python2.7/site-packages/ansible/plugins/callback/default.pyc
2018-06-20 14:54:00,483 p=23299 u=ansible |  PLAYBOOK: vmware.yml *******************************************************************************
2018-06-20 14:54:00,484 p=23299 u=ansible |  1 plays in vmware.yml
2018-06-20 14:54:00,501 p=23299 u=ansible |  PLAY [192.168.23.40] *******************************************************************************
2018-06-20 14:54:00,505 p=23299 u=ansible |  META: ran handlers
2018-06-20 14:54:00,509 p=23299 u=ansible |  TASK [Create a PulseConnectSecureVM from the template] *********************************************
2018-06-20 14:54:00,510 p=23299 u=ansible |  task path: /home/ansible/vmware.yml:4
2018-06-20 14:54:00,693 p=23299 u=ansible |  Using module file /usr/lib/python2.7/site-packages/ansible/modules/cloud/vmware/vmware_guest.py
2018-06-20 14:57:42,349 p=23299 u=ansible |  fatal: [192.168.23.40 -> localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "annotation": null,
            "cdrom": {},
            "cluster": null,
            "customization": {},
            "customvalues": [],
            "datacenter": "Datacenter",
            "disk": [
                {
                    "datastore": "datastore1",
                    "size_gb": 40
                }
            ],
            "esxi_hostname": null,
            "folder": "/bla/bla/",
            "force": false,
            "guest_id": null,
            "hardware": {},
            "hostname": "192.168.23.40",
            "is_template": false,
            "linked_clone": false,
            "name": "PulseConnectSecure",
            "name_match": "first",
            "networks": [
                {
                    "name": "VM Network",
                    "type": "dhcp"
                },
                {
                    "name": "VM Network 2",
                    "type": "dhcp"
                },
                {
                    "name": "VM Network",
                    "type": "dhcp"
                }
            ],
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "port": 443,
            "resource_pool": null,
            "snapshot_src": null,
            "state": "poweredon",
            "template": "Pulse Connect Secure",
            "username": "administrator@vsphere.local",
            "uuid": null,
            "validate_certs": false,
            "wait_for_ip_address": false
        }
    },
    "msg": "Failed to create a virtual machine : Customization of the guest operating system 'otherGuest64' is not supported in this configuration. Microsoft Vista (TM) and Linux guests with Logical Volume Manager are supported only for recent ESX host and VMware Tools versions. Refer to vCenter documentation for supported configurations."
}
2018-06-20 14:57:42,350 p=23299 u=ansible |     to retry, use: --limit @/home/ansible/vmware.retry

2018-06-20 14:57:42,350 p=23299 u=ansible |  PLAY RECAP *****************************************************************************************
2018-06-20 14:57:42,351 p=23299 u=ansible |  192.168.23.40              : ok=0    changed=0    unreachable=0    failed=1
```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/cloud/vmware/vmware_guest.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/vmware/vmware_guest.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @Akasurde @dav1x @nerzhul @pdellaert @warthog9
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by cigamit** [comment]
I am trying to find it, but I remember seeing a bug the other day that showed that if a NIC was set to DHCP (which is does by default) then it still tries to do customization even though it shouldn't technically.  There was a 2 line code fix attached to it.

**Comment by BarresB** [comment]
I went looking for that DHCP error you mentioned and found this:
https://github.com/ansible/ansible/pull/40990/commits/2103c904bfda48bef18a00d67b85c3fb0fa6113e --> https://github.com/ansible/ansible/pull/40990
Is this what you meant? Found it here: https://github.com/ansible/ansible/issues/40989
Didn't know the error is already known... sorry, hadn't found anything before opening the issue.
I'm not too familiar with GitHub, but when is it going to be implemented? Could be me, but I can't find something pointing to a version or a date.

**Comment by EricThiel** [comment]
Actually, the issue that will probably address this is https://github.com/ansible/ansible/issues/40637 - Mine (40990) only fixes it to not force customization for start_connected. Fixing DHCP was a bit more involved, so I did not tackle that one. In my install, I just manually rolled back https://github.com/ansible/ansible/pull/38518 and then when applying my fix for start_connected it worked for me.
